packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -1438,14 +1449,41 @@ export class DATE extends AbstractDataType<AcceptedDate> {
       return value;
     }
 
-    if (typeof value === 'string' || typeof value === 'number') {
+    if (typeof value === 'number') {
+      return new Date(value);
+    }
+
+    if (typeof value === 'string') {
+      if (!/\d{4}(?:-\d{2}){0,2}(?:(?:T| )\d{2}(?::\d{2}){2}){0,1}(?:\.\d{1,}){0,1}(?:Z|[+-]\d{2}(?::\d{2}){0,1}){0,1}/.test(value)) {
```

Comment:
nit: is it possible to extract these regexes out into a named constant? A bit tough to understand these :)

Reply:
I've moved these now.
LMK if they don't make sense.